### PR TITLE
list-ops: update test and version file

### LINKED ISSUE #1677: list-ops: update tests and version file (v2.3.0 -> v2.4.0)
The tests for the [list-ops exercise](https://github.com/exercism/java/tree/master/exercises/list-ops/src/test/java) are not up-to-date with the [canonical data](https://github.com/exercism/problem-specifications/tree/master/exercises/list-ops/canonical-data.json).
To check what has changed and if new tests have been added to the canonical data, please have a look at the [commit history](https://github.com/exercism/problem-specifications/commits/master/exercises/list-ops/canonical-data.json) of the canonical data file for the exercise list-ops.
Please update version and/or test file, and, if the new test(s) fail, fix the reference implementation.
Feel free to leave a comment if you have any questions. 

Background info: each exercise which has canonical data should have a version file. This file states which version of the canonical data the exercise implements. The version can be found at the top of the [canonical data file](https://github.com/exercism/problem-specifications/tree/master/exercises/list-ops/canonical-data.json) for that exercise.

### PR DESCRIPTION
**Pull request by lemoncurry**
closes #1677

---

Reviewer Resources:

[Track Policies](https://github.com/exercism/java/blob/master/POLICIES.md#event-checklist)

### CODE REVIEW

File: exercises/list-ops/src/test/java/ListOpsTest.java [hunk 1]
```diff
@@ -211,4 +211,23 @@ public void testReversingNonEmptyList() {
...
+                Collections.emptyList(),
+                Arrays.asList('4', '5', '6'));
+
+        assertEquals(
+                Arrays.asList(
+                        Arrays.asList('4', '5', '6'),
+                        Collections.emptyList(),
+                        Collections.singletonList('3'),
+                        Arrays.asList('1', '2')
+                ),
```
Comment (FridaTveit) [inline exercises/list-ops/src/test/java/ListOpsTest.java]: Very minor thing, but I think it's good to be consistent about either having the closing bracket on the same line or on a new line. I personally prefer on a new line, as I think it's more readable, but many people prefer it on the same line. Either is fine, as long as it's consistent.

So I'd prefer if we either move this closing bracket to the previous line, or change this file to have all closing brackets on a new line (though only when expressions are split over several lines of course). I have a slight preference for the second option, but either is fine. Does that make sense? 🙂

Comment (lemoncurry) [reply exercises/list-ops/src/test/java/ListOpsTest.java]: I changed it to new line.

File: exercises/list-ops/src/test/java/ListOpsTest.java [hunk 2]
```diff
@@ -137,7 +147,8 @@ public void testFoldLeftOnEmptyList() {
                 ListOps.foldLeft(
                     Collections.<Double>emptyList(),
                     2.0,
-                    (x, y) -> x * y));
+                    (x, y) -> x * y)
```
Comment (FridaTveit) [inline exercises/list-ops/src/test/java/ListOpsTest.java]: Technically the closing bracket at the end of this line should also be on a new line, to be completely consistent, as that closes the `foldLeft` which is spread over several lines. The same applies to the other tests using `foldLeft` and `foldRight` spread over several lines. It's not very important though, so I'm happy for this to be merged as it is if you prefer 🙂

Comment (lemoncurry) [reply exercises/list-ops/src/test/java/ListOpsTest.java]: Ah yes, I missed it. Now all the cases should be covered.

**Comment by lemoncurry** [comment]
Thank you for all your reviews @FridaTveit !

**Pull request merged**